### ISSUE TITLE
Port forwarding to host not working when using non default network

### ISSUE THREAD

**Issue by espenaf** [containers/libpod#5315]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Port forwarding to a container is not accessible locally or remote when using a newly created network via "podman network create".

**Steps to reproduce the issue:**

1. Fresh install of CentOS 8.1.1911 dvd iso with "dnf update" to latest version

2. podman create network podman2

3. podman run -dt -p 8080:8080/tcp --network=podman2 -e HTTPD_VAR_RUN=/var/run/httpd -e HTTPD_MAIN_CONF_D_PATH=/etc/httpd/conf.d -e HTTPD_MAIN_CONF_PATH=/etc/httpd/conf -e HTTPD_CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/httpd/ registry.fedoraproject.org/f29/httpd /usr/bin/run-httpd

4. curl http://localhost:8080

**Describe the results you received:**

Curl cannot connect and just hangs/timeouts

**Describe the results you expected:**

Success connection to httpd on 8080 port forwarded to newly created container.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
podman version 1.6.4
```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.12.12
  podman version: 1.6.4
host:
  BuildahVersion: 1.12.0-dev
  CgroupVersion: v1
  Conmon:
    package: conmon-2.0.6-1.module_el8.1.0+272+3e64ee36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.6, commit: 7a4f0dd7b20a3d4bf9ef3e5cbfac05606b08eac0'
  Distribution:
    distribution: '"centos"'
    version: "8"
  MemFree: 65781760
  MemTotal: 1389760512
  OCIRuntime:
    name: runc
    package: runc-1.0.0-64.rc9.module_el8.1.0+272+3e64ee36.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.1-dev'
  SwapFree: 1531179008
  SwapTotal: 2147479552
  arch: amd64
  cpus: 2
  eventlogger: journald
  hostname: localhost.localdomain
  kernel: 4.18.0-147.5.1.el8_1.x86_64
  os: linux
  rootless: false
  uptime: 4h 2m 56.19s (Approximately 0.17 days)
registries:
  blocked: null
  insecure: null
  search:
  - registry.centos.org
  - docker.io
store:
  ConfigFile: /etc/containers/storage.conf
  ContainerStore:
    number: 1
  GraphDriverName: overlay
  GraphOptions: {}
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 1
  RunRoot: /var/run/containers/storage
  VolumePath: /var/lib/containers/storage/volumes

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-1.6.4-2.module_el8.1.0+272+3e64ee36.x86_64
iptables-1.8.2-16.el8.x86_64
firewalld-0.7.0-5.el8.noarch
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Content of /etc/cni/net.d/podman2.conflist created by podman network create:

```
{
   "cniVersion": "0.4.0",
   "name": "podman2",
   "plugins": [
      {
       	 "type": "bridge",
         "bridge": "cni-podman1",
         "isGateway": true,
         "ipMasq": true,
         "ipam": {
            "type": "host-local",
            "routes": [
               {
                  "dst": "0.0.0.0/0"
               }
            ],
            "ranges": [
               [
                  {
                     "subnet": "10.89.0.0/24",
                     "gateway": "10.89.0.1"
                  }
               ]
            ]
         }
      },
      {
       	 "type": "portmap",
         "capabilities": {
            "portMappings": true
         }
      },
      {
       	 "type": "firewall",
         "backend": "iptables"
      }
   ]
}
```
Default network config /etc/cni/net.d/87-podman-bridge.conflist

```
{
    "cniVersion": "0.4.0",
    "name": "podman",
    "plugins": [
        {
            "type": "bridge",
            "bridge": "cni-podman0",
            "isGateway": true,
            "ipMasq": true,
            "ipam": {
                "type": "host-local",
                "routes": [
                    {
                     	"dst": "0.0.0.0/0"
                    }
                ],
                "ranges": [
                    [
                     	{
                            "subnet": "10.88.0.0/16",
                            "gateway": "10.88.0.1"
                        }
                    ]
                ]
            }
	},
	{
            "type": "portmap",
            "capabilities": {
                "portMappings": true
            }
	},
	{
            "type": "firewall",
            "backend": "iptables"
        }
    ]
}
```

Tested both on physical and virtual machine via libvirt/KVM. Also tested by changing backend to firewalld in podman2.conflist configuration, same issue.

When using the default podman network iptables is updated with the following:

```
Chain CNI-DN-b0eec3398ba92277030f4 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 CNI-HOSTPORT-SETMARK  tcp  --  *      *       10.88.0.3            0.0.0.0/0            tcp dpt:8080
    0     0 CNI-HOSTPORT-SETMARK  tcp  --  *      *       127.0.0.1            0.0.0.0/0            tcp dpt:8080
    0     0 DNAT       tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:8080 to:10.88.0.3:8080
```

When using podman2 network, this iptables rules are never created, full output from "iptables -t nat -nvL"

```
Chain PREROUTING (policy ACCEPT 1 packets, 328 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 CNI-HOSTPORT-DNAT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain POSTROUTING (policy ACCEPT 58 packets, 4443 bytes)
 pkts bytes target     prot opt in     out     source               destination         
  165 11256 CNI-HOSTPORT-MASQ  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* CNI portfwd requiring masquerade */
    3   210 RETURN     all  --  *      *       192.168.122.0/24     224.0.0.0/24        
    0     0 RETURN     all  --  *      *       192.168.122.0/24     255.255.255.255     
   61  3660 MASQUERADE  tcp  --  *      *       192.168.122.0/24    !192.168.122.0/24     masq ports: 1024-65535
  167 12205 MASQUERADE  udp  --  *      *       192.168.122.0/24    !192.168.122.0/24     masq ports: 1024-65535
    0     0 MASQUERADE  all  --  *      *       192.168.122.0/24    !192.168.122.0/24    
    0     0 CNI-ca9ef9d294b2b95a1ce194b0  all  --  *      *       10.89.0.2            0.0.0.0/0            /* name: "podman2" id: "a4dff5a59ae2fb6cca3adeecbd135cc537564d43c41d2f8eb3fc6d8ce9b7bdbb" */

Chain OUTPUT (policy ACCEPT 286 packets, 20308 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    4   240 CNI-HOSTPORT-DNAT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain CNI-HOSTPORT-SETMARK (0 references)
 pkts bytes target     prot opt in     out     source               destination         
    2   120 MARK       all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* CNI portfwd masquerade mark */ MARK or 0x2000

Chain CNI-HOSTPORT-MASQ (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    2   120 MASQUERADE  all  --  *      *       0.0.0.0/0            0.0.0.0/0            mark match 0x2000/0x2000

Chain CNI-HOSTPORT-DNAT (2 references)
 pkts bytes target     prot opt in     out     source               destination         

Chain CNI-ca9ef9d294b2b95a1ce194b0 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.89.0.0/24         /* name: "podman2" id: "a4dff5a59ae2fb6cca3adeecbd135cc537564d43c41d2f8eb3fc6d8ce9b7bdbb" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman2" id: "a4dff5a59ae2fb6cca3adeecbd135
```

podman port shows:

```
8080/tcp -> 0.0.0.0:8080
```

**Comment by mheon** [comment]
Can you try removing the following from the network definition:

```
	{
            "type": "firewall",
            "backend": "iptables"
        }
```

(You'll also need to remove the trailing `,` on the line before)

Then run a `firewall-cmd --reload` (assuming you use firewalld) and try running a container

**Comment by espenaf** [comment]
> Can you try removing the following from the network definition:
> 
> ```
> 	{
>             "type": "firewall",
>             "backend": "iptables"
>         }
> ```
> 
> (You'll also need to remove the trailing `,` on the line before)
> 
> Then run a `firewall-cmd --reload` (assuming you use firewalld) and try running a container

Firewalld is in use per default CentOS install. The change did not help, also tried to do a full reboot, but still not working.

**Comment by mheon** [comment]
@mccv1r0 PTAL - seems like the portmap plugin is not running, despite being in the CNI config?

**Comment by rhatdan** [comment]
@mheon @espenaf @mccv1r0 Do we know if this is still an issue?

**Comment by mccv1r0** [comment]
We don't allow remote machines to access podman by design.  As part of determining what would be needed to allow remote access, on rel8 @mheon @baude and I tested that this works (and I did on fedora 30) we verified that access from the local system `curl http://localhost:8080` worked.  If there is a problem, it's probably release specific.

**Comment by mccv1r0** [comment]
I've been able to reproduce this on fedora 30.

When `--network`is not the default network, CNI `portmap` plugin isn't supplied anything.  Put another way, -`-p 8080:8080` is supplied to the CNI `portmap`only for the default network, either when explicitly set via `--network` or not.  Using `--network=foo`results in CNI isn't supplied anyting.

I have only tested with `--network` omitted, `--network="podman"` (where "podman" is the name of the default network) and `--network="foo"` (where "foo" is not the default network.)   I haven't tried with `--network="podman","foo","bar"` (e.g. multiple networks) but I'm guessing the same results would be seen.

**Comment by snelders** [comment]
> ....Put another way, -`-p 8080:8080` is supplied to the CNI `portmap`only for the default network, either when explicitly set via `--network` or not. Using `--network=foo`results in CNI isn't supplied anyting.

How did you check that this is the case / where the bug is?

**Comment by snelders** [comment]
Reproduced this bug on RHEL 8.1, RHEL 8.2 Beta and CentOS 8.1

All using:
podman: 1.6.4
nftables nftables-0.9.0
firewalld (N.A. since disabled / but nftables enabled)

**Comment by guardiande** [comment]
Just noticed I opened a duplicate with #6065 :)

With CentOS 8.1 the problem still persists but I was unable to reproduce it on Fedora Server 32.

**Comment by rhatdan** [comment]
Should be fixed in podman-1.9.3
